src/Esfa.Vacancy.Register.Application/Queries/GetVacancy/GetVacancyQueryHandler.cs

```diff
@@ -35,6 +40,21 @@ public class GetVacancyQueryHandler : IAsyncRequestHandler<GetVacancyRequest, Ge
 
             if (vacancy == null) throw new ResourceNotFoundException($"Vacancy: {message.Reference}");
 
+            if (vacancy.FrameworkCode.HasValue)
+            {
+                var framework = await _trainingDetailService.GetFrameworkDetails(vacancy.FrameworkCode.Value);
```

Comment:
@sachpatel this was intentionally decoupled from the external api (ie. following the code architecture)